src/core/annotation.js

```diff
@@ -1013,6 +1016,32 @@ class PolygonAnnotation extends PolylineAnnotation {
   }
 }
 
+class InkAnnotation extends Annotation {
+  constructor(parameters) {
+    super(parameters);
+
+    this.data.annotationType = AnnotationType.INK;
+
+    let dict = parameters.dict;
+    let rawInkLists = dict.getArray('InkList');
+    this.data.inkLists = [];
+    for (let i = 0, ii = rawInkLists.length; i < ii; ++i) {
+      // The raw ink lists array contains arrays of numbers representing
+      // the alternating horizontal and vertical coordinates, respectively,
+      // of each vertex. Convert this to an array of objects with x and y
+      // coordinates.
+      this.data.inkLists.push([]);
+      for (let j = 0, jj = rawInkLists[i].length; j < jj; j += 2) {
+        this.data.inkLists[i].push({
+          x: rawInkLists[i][j],
+          y: rawInkLists[i][j + 1],
```

Comment:
Either of these could theoretically be indirect objects, and that needs to be handled to prevent future bugs (since `getArray` for good reason only dereferences top level elements).
So define `const xref = parameters.xref;` *above* the loops, and then change these lines to:
```javascript
  x: xref.fetchIfRef(rawInkLists[i][j]),
  y: xref.fetchIfRef(rawInkLists[i][j + 1]),
```